realm/realm-library/src/objectServer/java/io/realm/mongodb/AppConfiguration.java

```diff
@@ -556,10 +589,47 @@ public Builder defaultSyncErrorHandler(SyncSession.ErrorHandler errorHandler) {
          * the {@link io.realm.mongodb.sync.SyncConfiguration}.
          *
          * @param handler the default Client Reset handler.
+         *                
+         * @deprecated replaced by {@link #defaultSyncClientResetStrategy(ManuallyRecoverUnsyncedChangesStrategy)}
          */
+        @Deprecated
         public Builder defaultClientResetHandler(SyncSession.ClientResetHandler handler) {
             Util.checkNull(handler, "handler");
-            defaultClientResetHandler = handler;
+            defaultSyncClientResetStrategy = handler;
+            return this;
+        }
+
+        /**
+         * Sets the default sync client reset strategy used by Synced Realms when they report a Client Reset.
+         * session.
+         * <p>
+         * This default can be overridden by calling
+         * {@link io.realm.mongodb.sync.SyncConfiguration.Builder#syncClientResetStrategy(ManuallyRecoverUnsyncedChangesStrategy)}
+         * or {@link io.realm.mongodb.sync.SyncConfiguration.Builder#syncClientResetStrategy(DiscardUnsyncedChangesStrategy)}
+         * when creating the {@link io.realm.mongodb.sync.SyncConfiguration}.
+         *
+         * @param strategy the default sync client reset strategy.
+         */
+        public Builder defaultSyncClientResetStrategy(@Nonnull ManuallyRecoverUnsyncedChangesStrategy strategy) {
```

Comment:
Why do you have two builder methods with `ManuallyRecoerUnsyncedChangesStrategy` and `DiscardUnsynchedChangesStrategy` instead of just one with `SyncClientResetStrategy` and then mention the two variants?

Reply:
Because a SyncClientResetStrategy is not valid strategy, and it would allow users to implement an empty strategy. Overloading the reset forces the users to pass a valid one: `ManuallyRecoverUnsyncedChangesStrategy` or `DiscardUnsyncedChangesStrategy`.